src/bp/core/config/config-writer.ts

```diff
@@ -74,29 +90,42 @@ export class BotConfigWriter {
     }
   }
 
-  private async _writeToFile(config: BotConfig) {
-    const filePath = this.getBotFilePath(config.id)
+  private async _writeConfig(botId: string, config: BotConfig) {
     const fileName = 'bot.config.json'
-
-    try {
-      await fse.ensureDir(filePath)
-      await fse.writeFile(filePath + fileName, JSON.stringify(config, undefined, 2))
-    } catch (e) {
-      throw new VError(e, `Error writing file "${filePath}"`)
-    }
+    const scopedGhost = this.ghost.forBot(botId)
+    await scopedGhost.upsertFile('/', fileName, JSON.stringify(config, undefined, 2))
   }
 
   async deleteBotFolder(botId: string) {
-    const filePath = this.getBotFilePath(botId)
+    const scopedGhost = this.ghost.forBot(botId)
+    await scopedGhost.deleteFile('/', 'bot.config.json')
+  }
 
-    try {
-      await fse.remove(filePath)
-    } catch (err) {
-      throw new VError(err, `Error while deleting file at "${filePath}"`)
+  private _listDir(
```

Comment:
If I understand right, what ou did here is a filtered implementation of directory walk (recursive directory listing).
Technically you could have used the node-klaw package which is a fse implementation of walk then filter the results starting with a `.`

I guess keeping the code as is is fine. If we ever feel the need to reuse this, either we move it in a utility package or we use node-klaw

Reply:
didn't know about klaw though. didn't check for a lib that does that :P

Reply:
@rndlaine well it was included in [fs-extra](https://github.com/jprichardson/node-fs-extra) but moved to [klaw](https://github.com/jprichardson/node-klaw) & [klaw-sync](https://github.com/manidlou/node-klaw-sync) for an unknown reason ..